### DISCUSSION TITLE
리팩토링 리스트

### DISCUSSION THREAD

**Discussion by UJ15** [prgrms-web-devcourse/Team-Saiko-BidMarket-BE#78] [General]
## 회원
- [ ] 로그인 Access Token, refresh Token 도입(redis)
- [ ] jwtAuthenticationFilter 리팩토링
- [ ] successHandler 리팩토링 (JWT 토큰 쿠키에 저장해서 넘겨주는 방식으로)
- [ ] 복잡한 회원 권한 로직 제거, 권한 enum type으로 변경, admin 제거
- [ ] 다른 사람 id로 로그인 되어있는 이슈 해결

## 채팅
- [ ] 채팅 메세지 저장 방식 변경 (Redis)
- [ ] 채팅메시지 stomp 헤더 토큰 검증

## 신고
- [ ] 현재 유저 중복신고가 불가함 하지만 게시글, 댓글 신고시 유저가 같이 신고되는데 이슈 발견
    - 유저 -> 게시글, 댓글 순으로 신고가 되면 문제가 없지만 게시글, 댓글 신고가 먼저 될경우 해당 유저 신고가 중복신고로 불가능하게됨

## 구조
- [ ] (Optional) 멀티 모듈 적용 -> 코어, 배치, api 나눠서 , 테스트 병렬처리로 속도 개선
- [ ] 문서화
    - [ ] 컨벤션 문서화
    - [ ] API 문서화 (RestDocs)
    - [ ] 이슈 문서화 (백엔드 문서에 있는 내용들)
- [ ] 처리 되지않는 runtime exception 잡기
- [ ] 로깅 전략 수립, logback

## Spring Batch
- [ ] S3 안쓰는 이미지 제거 구현

## 쿼리 DSL
- [ ] Page 객체 만들기

## CI/CD
- [ ] 운영, 개발서버 배포그룹 분리
- [ ] docker 컨테이너화
- [ ] sonarqube -> sonarcloud

## 기타
- [ ] 부하 테스트 해보고 개선해보기
- [ ] 소나 큐브 정적 코드분석 결과에 따라 문제 되는 부분 수정필요
- [ ] 테스트 속도 개선

**Comment by dojinyou** [comment]
id encoder 없애고 id로 정리하였습니다! 

관련 pr : #90
